File: flutter/lib/model/PostModel.dart
import 'package:facebook/model/UserInfor.dart';

class PostModel {
  int like;
  int comment;
  List<dynamic> images;
  bool is_liked;
  List<dynamic> video;
  String status;
  bool modified;
  String id;
  UserChatInfor author;
  DateTime created_at;
  String described;
  PostModel(
      {this.like,
      this.comment,
      this.images,
      this.is_liked,
      this.video,
        this.status,
      this.modified,
      this.id,this.author,this.created_at,this.described});

  factory PostModel.fromJson(Map<String, dynamic> json) {
    return PostModel(
        like: json['like'],
        comment: json['comment'],
        images: json['images'],
        is_liked: json['is_liked'],
        video: json['video'],
        status : json['status'],
        modified: json['modified'],
        id: json['id'].toString(),
        author: UserChatInfor.fromJson(json["author"]),
        created_at:DateTime.parse(json["created_at"]),
        described:json["described"]);
  }
}

class SingleChat {
  String user;
  String mess;
  SingleChat({this.mess, this.user});
}

File: server/models/post/postModel.js
'use strict';

let mongoose = require('../../db');
let autoIncrement = require('mongoose-auto-increment');
let Schema = mongoose.Schema;

let PostSchema = new Schema({
    id: {
        type: Number,
        required: true,
    },
    described: {
        type: String,
        default: null,
    },
    like: {
        type: Number,
        default: 0
    },
    comment: {
        type: Number,
        default: 0
    },
    images: {
        type: Array,
        default: null
    },
    is_liked: {
        type: Boolean,
        default: false
    },
    video: {
        type: Array,
        default: null
    },
    author: {
        type: Object,
        required: true
    },
    status: {
        type: String,
        default: null
    },
    modified: {
        type: Boolean,
        required: true,
        default: true
    },
    is_blocked: {
        type: Boolean,
        required: true,
        default: false
    },
    can_edit: {
        type: Boolean,
        required: true,
        default: true
    },
    banned: {
        type: Boolean,
        required: true,
        default: false
    },
    can_comment: {
        type: Boolean,
        required: true,
        default: true
    },
    created_at: {
        type: Date,
        default: Date.now
    }
});

PostSchema.plugin(autoIncrement.plugin, {
    model: 'Post',
    field: 'id'
});

module.exports = mongoose.model('Post', PostSchema);